File: apps/gustav/src/features/onboarding/domain/weekday.ts
// Day of the week — used for `trainingDays`/`planningDay`. The wire value
// (`infra/supabase/migrations/0001_init.sql`) is the full English word,
// matching the planner's `Weekday` domain type
// (`_shared/planner/models/enums.ts`) directly — no translation at the
// boundary.
export const WEEKDAY_ORDER = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday'] as const;
export type Weekday = (typeof WEEKDAY_ORDER)[number];

const LABEL_BY_WEEKDAY: Record<Weekday, string> = {
  monday: 'Mo',
  tuesday: 'Di',
  wednesday: 'Mi',
  thursday: 'Do',
  friday: 'Fr',
  saturday: 'Sa',
  sunday: 'So',
};

const FULL_LABEL_BY_WEEKDAY: Record<Weekday, string> = {
  monday: 'Montag',
  tuesday: 'Dienstag',
  wednesday: 'Mittwoch',
  thursday: 'Donnerstag',
  friday: 'Freitag',
  saturday: 'Samstag',
  sunday: 'Sonntag',
};

export function weekdayShortLabel(day: Weekday): string {
  return LABEL_BY_WEEKDAY[day];
}

export function weekdayFullLabel(day: Weekday): string {
  return FULL_LABEL_BY_WEEKDAY[day];
}

File: apps/gustav/lib/features/onboarding/domain/weekday.dart
/// Day of the week — used for `trainingDays`/`planningDay`. The German wire
/// value (`content/*.yaml`, `infra/supabase/migrations/0001_init.sql`) is
/// the two-letter abbreviation; see `toGerman`/`weekdayFromGerman`.
enum Weekday { monday, tuesday, wednesday, thursday, friday, saturday, sunday }

const _germanByWeekday = {
  Weekday.monday: 'mo',
  Weekday.tuesday: 'di',
  Weekday.wednesday: 'mi',
  Weekday.thursday: 'do',
  Weekday.friday: 'fr',
  Weekday.saturday: 'sa',
  Weekday.sunday: 'so',
};

extension WeekdayGerman on Weekday {
  String toGerman() => _germanByWeekday[this]!;
}

Weekday weekdayFromGerman(String value) {
  return _germanByWeekday.entries.firstWhere((e) => e.value == value).key;
}
